design/ceph/object/store.md

```diff
@@ -351,3 +351,285 @@ spec:
+
+```yaml
+# Default Data pool
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: default.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Default index pool
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: default.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+# Bucket Data pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket Data pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER` and `REDUCED_REDUNDANCY`:
+> [!NOTE]  
+> Ceph allows arbitrary name for StorageClasses, however some clients/libs insist on AWS names so it is recommended to use one of the [valid x-amz-storage-class values](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) for better compatibility:
+>
+> `STANDARD | REDUCED_REDUNDANCY | STANDARD_IA | ONEZONE_IA | INTELLIGENT_TIERING | GLACIER | DEEP_ARCHIVE | OUTPOSTS | GLACIER_IR | SNOW | EXPRESS_ONEZONE`
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: reduced.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+```
+
+Then update `CephObjectStore` CRD  `sharedPools` section with target placements and StorageClasses definitions:
+
+```yaml
+spec:
+  # [OPTIONAL] - sharedPools is mutually exclusive with spec.metadataPool and spec.dataPool
+  # It is the place to tell RGW which pools can be used to store buckets and objects data and metadata.
+  # All pools under this section should be created beforehand.
+  # Rook will append rados namespace to each of the pools, so it is safe to use the same pool across multiple RGW instances.
+  # It is also safe to use the same pool in different target placements.
+  sharedPools: 
+    #  [REQUIRED] - metadata pool. Will be used by default to store bucket index
+    metadataPoolName: default.rgw.buckets.index # results to default.rgw.buckets.index:<rgw_name>.index
+
+    #  [REQUIRED] - data pool. Will be used by default to store bucket objects
+    dataPoolName: default.rgw.buckets.data # results to default.rgw.buckets.data:<rgw_name>.data
+
+    #  [OPTIONAL] - non erasure coded data pool. Will be used by default to store bucket multipart objects
```

Comment:
What about putting these new properties under `placement`? Something like:
```yaml
placement:
  dataNonECPoolName: ""
  storageClasses: 
      - name: GLACIER
        dataPoolName: glacier.rgw.buckets.data
```

Reply:
Good question. I was thinking about naming but don't have a good solution. I've came up with 2 approaches:

1. Follow `zonegroup` and `zone` placement json structure. It will be consistent with Ceph `radosgw-admin` CLI and Ceph docs and so easy to understand for Ceph users. 
   ```yaml
   spec:
     poolPlacement:
       default:
         dataPool: ""
         indexPool: ""
         dataExtraPool: ""
         storageClasses:
          - name: GLACIER
            dataPool: ""
        additional:
         hot:
           dataPool: ""
           indexPool: ""
           dataExtraPool: ""
           storageClasses:
            - ...
          ...
   ```
2. Hide Ceph complexity from rook user. Maybe it will be more user-friendly to use S3 or SWIFT terms and concepts in ObjectStore CRDs:
   ```yaml
   spec:
     locations:
       default:
         dataPool: ""
         metadataPool: ""
         storageClasses:
          - name: GLACIER
            dataPool: ""
       additional:
         hot:
           dataPool: ""
           metadataPool: ""
           storageClasses:
             - ...
            ...
   ```

As a result, i've tried to merge ceph json naming (approach 1.) with existing `sharedPools` naming. In this case `dataNonECPoolName` should be on the same level with `dataPool` and `metadataPool`. But i am not sure that this is the best approach. What do you think?

Reply:
+1 for `poolPlacement`

Reply:
Agreed on `poolPlacement`

Reply:
i am a little bit confused. should i use `poolPlacement` instead of `sharedPools` or should i use  `poolPlacement` for additional pool placements under `sharedPools`?

Reply:
How about this example? So this feature won't modify any existing properties, only add the new settings. For example:
```yaml
spec:
  sharedPools:
    metadataPoolName: rgw-meta-pool
    dataPoolName: rgw-data-pool
    preserveRadosNamespaceDataOnDelete: true
    # new settings
    poolPlacements:
    - name: "default" # The default placement is optional in case of overriding the main pools above
      dataPool: ""
      indexPool: ""
      dataExtraPool: ""
      storageClasses:
      - name: GLACIER
         dataPool: ""
    - name: "hot"
      dataPool: ""
      indexPool: ""
      dataExtraPool: ""
      storageClasses:
      - ...
```
Also, a few other thoughts to consider in the above example:
- No "additional" section. All of the poolPlacements are included there with a name, and one of them has a special name "default" that will be treated as the default.
- If the "default" poolPlacement is not defined, the default placement could be the main pools listed under `sharedPools`?

Reply:
I still dislike suggesting that a user create a storageclass named `GLACIER`

Reply:
@travisn looks good! i will update design doc

Reply:
updated